tailnet/proto/tailnet.proto

```diff
@@ -99,7 +102,91 @@ message CoordinateResponse {
 	string error = 2;
 }
 
+message Netcheck {
+	bool UDP = 1;
+	bool IPv6 = 2;
+	bool IPv4 = 3;
+	bool IPv6CanSend = 4;
+	bool IPv4CanSend = 5;
+	bool OSHasIPv6 = 6;
+	bool ICMPv4 = 7;
+
+	google.protobuf.BoolValue MappingVariesByDestIP = 8;
+	google.protobuf.BoolValue HairPinning = 9;
+	google.protobuf.BoolValue UPnP = 10;
+	google.protobuf.BoolValue PMP = 11;
+	google.protobuf.BoolValue PCP = 12;
+
+	int32 PreferredDERP = 13;  // 0 for unknown
+
+	map<int32, google.protobuf.Duration> RegionLatency = 14;
+	map<int32, google.protobuf.Duration> RegionV4Latency = 15;
+	map<int32, google.protobuf.Duration> RegionV6Latency = 16;
+
+	string GlobalV4 = 17;
+	string GlobalV6 = 18;
+
+	google.protobuf.BoolValue CaptivePortal = 19;
+}
+
+message TelemetryRequest {
```

Comment:
Yeah, I was kinda debating back and forth with this when porting it to protobuf. I felt that since this is over DRPC anyways, the overhead of calling multiple times is more negligable than if this were an HTTP API like I originially thought. You're right though that the array is more flexible. I'll switch it back to that.